`win32_ucontext.h`, included by `julia.h` no longer available.

PackageCompiler CI started erroring with:

```
In file included from D:\a\PackageCompiler.jl\PackageCompiler.jl\src\embedding_wrapper.c:10:
C:\hostedtoolcache\windows\julia\nightly\x64\include\julia/julia.h:42:12: fatal error: win32_ucontext.h: No such file or directory
 #  include "win32_ucontext.h"
            ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

on nightly. This file was moved in https://github.com/JuliaLang/julia/pull/41936 from `src/support/win32_ucontext.h` to `src/win32_ucontext.h`.

--- a/src/Makefile
+++ b/src/Makefile
@@ -111,6 +111,9 @@ UV_HEADERS += uv.h
 UV_HEADERS += uv/*.h
 endif
 PUBLIC_HEADERS := $(BUILDDIR)/julia_version.h $(wildcard $(SRCDIR)/support/*.h) $(addprefix $(SRCDIR)/,julia.h julia_assert.h julia_threads.h julia_fasttls.h julia_locks.h julia_atomics.h jloptions.h)
+ifeq ($(OS),WINNT)
+PUBLIC_HEADERS += $(addprefix $(SRCDIR)/,win32_ucontext.h)
+endif
 HEADERS := $(PUBLIC_HEADERS) $(addprefix $(SRCDIR)/,julia_internal.h options.h timing.h) $(addprefix $(BUILDDIR)/,$(DTRACE_HEADERS) jl_internal_funcs.inc)
 PUBLIC_HEADERS += $(addprefix $(SRCDIR)/,julia_gcext.h)
 PUBLIC_HEADER_TARGETS := $(addprefix $(build_includedir)/julia/,$(notdir $(PUBLIC_HEADERS)) $(UV_HEADERS))

